### PR TITLE
Accessibility improvement for DetailsList while placeholder data is being displayed

### LINKED ISSUE #9421: Shimmer accessibility reads table has 11 rows when shimmer is in loading state
Issue: Using narrator for accessibility, when navigating to the first column, the narrator reads that table has 11 rows and 9 columns... This issue is when the shimmer is still in loading state. As there are 11 animations while loading narrator reads 11 rows which is not the actual case. I feel it should read  that the rows are being loaded. Once loaded, it reads properly the exact number of rows.

## Codepen demonstrating the issue
Navigate to https://developer.microsoft.com/en-us/fabric#/controls/web/detailslist/shimmer
Use the first example to and Export to Codepen for Repro of the issue

## Actual behavior:
When the shimmer loads, and when you navigate to the column, the narrator reads Table has 11 rows, 9 columns.... 

## Expected behavior:
As the table is being loaded it should read, Table has 9 colums(as they are pre-defined and visible) and rows are being loaded or something similar...

### PR DESCRIPTION
**Pull request by Rajeev-K**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #9421 
- [x] Include a change request file using `$ npm run change`

#### Description of changes

DetailsList accessibility now reports 0 rows while placeholder data is being displayed. Also added a prop to ShimmeredDetailsList to specify aria label (such as a "data is being loaded" message) while shimmer is being displayed.

#### Focus areas to test

DetailsList and ShimmeredDetailsList

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9484)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.base.tsx [hunk 1]
```diff
@@ -457,7 +458,7 @@ export class DetailsListBase extends BaseComponent<IDetailsListProps, IDetailsLi
                   onItemInvoked={onItemInvoked}
                   onItemContextMenu={onItemContextMenu}
                   enterModalOnTouch={this.props.enterModalSelectionOnTouch}
-                  {...selectionZoneProps || {}}
+                  {...(selectionZoneProps || {})}
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.base.tsx]: Good catch 👍

**Comment by KevinTCoughlin** [comment]
@Rajeev-K, you'll have to update snapshots for CI to pass. Current error is:

```shell
  ‌● ‌Component Examples › renders Shimmer.Application.Example.tsx correctly‌

    ‌expect(‌value‌).toMatchSnapshot()‌

    ‌Received value‌ does not match ‌stored snapshot &quot;Component Examples renders Shimmer.Application.Example.tsx correctly 1&quot;‌.‌

    ‌- Snapshot‌
    ‌+ Received‌

    ‌@@ -208,12 +208,13 @@‌
    ‌          data-automationid=&quot;DetailsList&quot;‌
    ‌          data-is-scrollable=&quot;false&quot;‌
    ‌        &gt;‌
    ‌          &lt;div‌
    ‌            aria-colcount={1}‌
    ‌+           aria-label=&quot;Content is being fetched&quot;‌
    ‌            aria-readonly=&quot;true&quot;‌
    ‌-           aria-rowcount={11}‌
    ‌+           aria-rowcount={0}‌
    ‌            role=&quot;grid&quot;‌
    ‌          &gt;‌
    ‌            &lt;div‌
    ‌              className=‌
    ‌                  ms-DetailsList-headerWrapper‌

    ‌ ‌ 193 | ‌              ‌const‌ component ‌=‌ renderer‌.‌create(‌&lt;‌ComponentUnderTest‌ ‌&#x2F;‌&gt;‌)‌;‌
    ‌ ‌ 194 | ‌              ‌const‌ tree ‌=‌ component‌.‌toJSON()‌;‌
    ‌&gt;‌ 195 | ‌              (expect(tree) as any)‌.‌toMatchSpecificSnapshot(componentFileName)‌;‌
    ‌ ‌     | ‌                                    ‌^‌
    ‌ ‌ 196 | ‌            })‌;‌
    ‌ ‌ 197 | ‌        } ‌catch‌ (e) {‌
    ‌ ‌ 198 | ‌          ‌&#x2F;&#x2F; If you are getting this error with an example file make sure that the example file only‌

      ‌at ‌src&#x2F;components&#x2F;ComponentExamples.test.tsx‌:195:37‌
          at Array.forEach (&lt;anonymous&gt;)‌
      ‌at Object.&lt;anonymous&gt; (‌src&#x2F;components&#x2F;ComponentExamples.test.tsx‌:189:14)‌

 › 1 snapshot failed.‌
<
```

To fix, run `npm run update-snapshots` at repo root and commit the result.

**Comment by KevinTCoughlin** [comment]
@Rajeev-K, do you think we should use `0` or `-1` for the Shimmer placeholder data scenario?

The W3 ARIA docs say the following:

>Authors MUST set the value of aria-rowcount to an integer equal to the number of rows in the full table. If the total number of rows is unknown, authors MUST set the value of aria-rowcount to -1 to indicate that the value should not be calculated by the user agent.

https://www.w3.org/TR/wai-aria-1.1/#aria-rowcount

Just a consideration. I'm unsure of the correct value in the Shimmer scenario. Both `0` and `-1` seem reasonable to me.

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.base.tsx (hunk 1 above)
Comment (Rajeev-K) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.base.tsx]: That change was made automatically by prettier!

**Comment by Rajeev-K** [comment]
I have changed this to -1 to comply with the W3 recommendation

---
In reply to: [502835762](https://github.com/OfficeDev/office-ui-fabric-react/pull/9484#issuecomment-502835762) [](ancestors = 502835762)

**Comment by Rajeev-K** [comment]
Snapshots have been updated.

---
In reply to: [502834864](https://github.com/OfficeDev/office-ui-fabric-react/pull/9484#issuecomment-502834864) [](ancestors = 502834864)

**Pull request merged**